### DISCUSSION TITLE
Does QUBO solver use one network by default?

### DISCUSSION THREAD

**Discussion by nnbtam** [lava-nc/lava-optimization#234] [Q&A] [answered]
Hi Lava team,

I'm using lava-optimization to solve QUBO problems, running on CPU and I usually see this sentence.

>Host: better solution found by network 0 at step 3 with cost -2: [0 0 0 1 0]

If my understanding is correct, does it mean it all variables of QUBO (in this case 5 variables) are solved in a parallel manner with only one SNN?

Thank you very much!
Tam

**Comment by tim-shea** [comment, marked as answer]
Yes, the QUBO solver uses a single SNN.

The structure of that SNN is one layer of spiking neurons with recurrent connections.

The solver iteratively finds better and better solutions, keeping track of the best solution found so far. This message is printed whenever the current solution improves on the previous best.

If you try running a larger problem of perhaps 50 - 100 variables, set the target_cost to -100, and set the timesteps to 1000 or so, you should see this message come up multiple times, getting incrementally better each time.
